spec/lib/services/workflow-api-service-spec.js

```diff
@@ -250,4 +279,85 @@ describe('Http.Services.Api.Workflows', function () {
             expect(persistStub).to.have.been.calledOnce;
         });
     });
+
+    it('should get workflows tasks by name', function () {
+        store.getTaskDefinitions.resolves({ injectableName: 'test' });
+        return workflowApiService.getWorkflowsTasksByName(taskDefinition)
+        .then(function() {
+            expect(store.getTaskDefinitions).to.have.been.calledOnce;
+            expect(store.getTaskDefinitions).to.have.been.calledWith(taskDefinition);
+        });
+    });
+
+
+    it('should delete/destroy graph', function () {
+        waterline.graphdefinitions.destroy.resolves(graph);
+        store.destroyGraphDefinition.resolves({ injectableName: 'test' });
+        return workflowApiService.destroyGraphDefinition(taskDefinition)
+        .then(function() {
+            expect(store.destroyGraphDefinition).to.have.been.calledOnce;
+            expect(store.destroyGraphDefinition).to.have.been.calledWith(taskDefinition);
+        });
+    });
+
+
+    it('should put workflows tasks by name', function () {
+        store.getTaskDefinitions.resolves(task);
+        return workflowApiService.putWorkflowsTasksByName(taskDefinition, task)
+        .then(function() {
+            expect(store.getTaskDefinitions).to.have.been.calledOnce;
+            expect(store.getTaskDefinitions).to.have.been.calledWith(task);
+            expect(store.persistTaskDefinition).to.have.been.calledOnce;
+            expect(store.persistTaskDefinition).to.have.been.calledWith(taskDefinition);
+        });
+    });
+
+    it('should throw error, when cancelling an non-active workflow ', function () {
+        var mockWorkflowError = new Errors.TaskCancellationError(
+            "testid is not an active workflow"
+        );
+        waterline.graphobjects.needByIdentifier.rejects(mockWorkflowError);
+        return workflowApiService.cancelTaskGraph()
+            .should.eventually.be.rejectedWith(mockWorkflowError);
```

Comment:
Yeah the former should work fine I believe. The second doesn't test the case where an error isn't thrown, so the first block is the better options.